core/playback/decoder.go

```diff
@@ -0,0 +1,47 @@
+package playback
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"os"
+
+	"github.com/faiface/beep"
+	"github.com/faiface/beep/flac"
+	"github.com/faiface/beep/mp3"
+	"github.com/faiface/beep/wav"
+	"github.com/navidrome/navidrome/core/ffmpeg"
+	"github.com/navidrome/navidrome/log"
+)
+
+func decodeMp3(path string) (s beep.StreamSeekCloser, format beep.Format, err error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, beep.Format{}, err
+	}
+	return mp3.Decode(f)
+}
+
+func decodeWAV(path string) (s beep.StreamSeekCloser, format beep.Format, err error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, beep.Format{}, err
+	}
+	return wav.Decode(f)
+}
+
+func decodeFLAC(ctx context.Context, path string) (s beep.StreamSeekCloser, format beep.Format, err error) {
+	fFmpeg := ffmpeg.New()
+	readCloser, err := fFmpeg.ConvertToFLAC(ctx, path)
+	if err != nil {
+		log.Error(err)
+		return
+	}
+
+	data, err := io.ReadAll(readCloser)
```

Comment:
@deluan-tc That's a tricky one since: The playback system BEEP needs to have a io.Seeker to postion the playback and seek into the track/song. FLAC streams are coming from the FFMPEG based transformer which are pipes. They are not seekable. Therefore i had to buffer them in a byte array to turn them into something seekable. The only way to circumvent this is to use ( and take care of ) a tempfile.

What should we do?

Reply:
If we absolutely need to Seek, then a temp file is the only solution here. We just have to be sure to trigger the transcoding way before it is actually needed, or the playback would stop waiting for the transcoding to finish.

Reply:
I've implemented a tempfile-version, since BEEP has to have a seeker:

"The Seek method will panic if r is not io.Seeker"

Triggering the transcoding upfront is only possible within a playlist (given by the set command). Otherwise i simply don't know what the user clicks next ...

I would propose to keep it as it is and do optimizing, caching, ahead of time transcoding in a separate story.